src/io/perun.clj

```diff
@@ -74,18 +73,43 @@
       (u/info "Remove draft files. Remaining %s files\n" (count updated-files))
       fs-with-meta)))
 
-(defn- create-filepath [file]
-  (assoc file :filepath (str  "/" (:filename file) "/index.html")))
+(defn ^:private default-slug-fn [filename]
+  (->> (string/split filename #"[-\.]")
+       (drop 3)
+       drop-last
+       (string/join "-")
+       string/lower-case))
+
+(deftask slug
+  "Adds :slug key to files metadata. Slug is derived from filename."
+  [s slug-fn SLUGFN code "Function to build slug from filename"]
+  (boot/with-pre-wrap fileset
+    (let [slug-fn       (or slug-fn default-slug-fn)
+          files         (:metadata (meta fileset))
+          updated-files (into {}
```

Comment:
You can use `https://github.com/hashobject/perun/blob/master/src/io/perun/core.clj#L36` function here. Will be a little bit shorter